### PR TITLE
CLN: Avoid bare pytest.raises in computation/test_eval.py 

### PR DESCRIPTION
**Pull request by Vlek**
* [x]  ref #30999
 
* [x]  tests added / passed
 
* [x]  passes `black pandas`

* [x]  passes `git diff origin/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/tests/computation/test_eval.py [hunk 1]
```diff
@@ -1593,19 +1610,21 @@ def test_simple_in_ops(self):
...
-                pd.eval("3 not in (1, 2)", engine=self.engine, parser=self.parser)
-            with pytest.raises(NotImplementedError):
+            with pytest.raises(NotImplementedError, match=msg):
                 pd.eval(
                     "[(3,)] in (1, 2, [(3,)])", engine=self.engine, parser=self.parser
                 )
-            with pytest.raises(NotImplementedError):
+            msg = "'NotIn' nodes are not implemented"
+            with pytest.raises(NotImplementedError, match=msg):
+                pd.eval("3 not in (1, 2)", engine=self.engine, parser=self.parser)
```
Comment (datapythonista) [inline pandas/tests/computation/test_eval.py]: Looks like those tests have been swapped. The order doesn't make a difference, it's not a problem, but the diff becames confusing, so pointing it out for other reviewers.

**Review by datapythonista — APPROVED** [review]
Thanks @Vlek nice work.

@gfyoung do you want to have a look?

File: pandas/tests/computation/test_eval.py (hunk 1 above)
Comment (gfyoung) [reply pandas/tests/computation/test_eval.py]: Some kind of parametrization would be good here, but another PR for another time.

**Review by gfyoung — APPROVED** [review]

**Comment by datapythonista** [comment]
Thanks @Vlek

**Pull request merged**